### ISSUE TITLE
Create a Link widget to better support accessibility

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#119846]
Currently Flutter relies on developers to put their url launching logic in the on press callback. This does not work well for accessibility. We cannot automatically apply semantics information to the semantic node since we don't know what happens in the onpress. The idea is to create a widget that takes in a url and open the link when user press the widget. For example something similar to this

```dart
Link(
  url: Uri.parse('www.google.com'),
  child: Text('I am a link'),
)
```

**Comment by chunhtai** [comment]
cc @mdebbar  I noticed the url_launcher is a plugin. so that means the widget cannot sit in framework. Do you think it can be move into framework along with url_launcher? I am not sure if that is a good idea though

**Comment by mdebbar** [comment]
@chunhtai there's already a Link widget in `url_launcher` and it takes a uri. Can we make it supply the extra semantics info needed for links? Something like this:
```dart
return MergeSemantics(
  Semantics(
    link: true,
    linkUri: uri,
    child: ...
  ),
);
```

We should do that here:
https://github.com/flutter/plugins/blob/main/packages/url_launcher/url_launcher/lib/src/link.dart#L143
and here:
https://github.com/flutter/plugins/blob/main/packages/url_launcher/url_launcher_web/lib/src/link.dart#L68

**Comment by chunhtai** [comment]
sounds good to me, updating title

**Comment by emirhalici** [comment]
This appears fixed in https://github.com/flutter/packages/pull/6711.

**Comment by mdebbar** [comment]
> This appears fixed in [flutter/packages#6711](https://github.com/flutter/packages/pull/6711).

That's right, thank you! Closing.
